lib/NavLink.js

```diff
@@ -32,6 +32,28 @@ NavLink = React.createClass({
         navParams: React.PropTypes.object,
         followLink: React.PropTypes.bool
     },
+    getInitialState: function () {
+        return {
+            href: this._getHrefFromProps(this.props)
+        };
+    },
+    componentWillReceiveProps: function (nextProps) {
+        this.setState({
+            href: this._getHrefFromProps(nextProps)
+        });
+    },
+    _getHrefFromProps: function (props) {
+        var href = props.href;
+        var routeName = props.routeName;
+        if (!href && routeName) {
+            var context = props.context || this.context;
+            href = context.makePath(routeName, props.navParams);
+        }
+        if (!href) {
+            throw new Error('NavLink created without href or routeName');
```

Comment:
or valid routeName...  user could've provided a routeName, which is not defined in app's routes.